hudi-gcp/src/main/java/org/apache/hudi/gcp/bigquery/HoodieBigQuerySyncClient.java

```diff
@@ -147,6 +159,31 @@ public void createManifestTable(String tableName, String sourceUri) {
     }
   }
 
+  /**
+   * Updates the schema for the given table if the schema has changed.
+   * @param tableName name of the table in BigQuery
+   * @param schema latest schema for the table
+   */
+  public void updateTableSchema(String tableName, Schema schema, List<String> partitionFields) {
+    Table existingTable = bigquery.getTable(TableId.of(projectId, datasetName, tableName));
+    ExternalTableDefinition definition = existingTable.getDefinition();
+    Schema remoteTableSchema = definition.getSchema();
+    // Add the partition fields into the schema to avoid conflicts while updating
+    List<Field> updatedTableFields = remoteTableSchema.getFields().stream()
+        .filter(field -> partitionFields.contains(field.getName()))
+        .collect(Collectors.toList());
+    updatedTableFields.addAll(schema.getFields());
+    Schema finalSchema = Schema.of(updatedTableFields);
+    if (definition.getSchema() != null && definition.getSchema().equals(finalSchema)) {
+      return; // No need to update schema.
+    }
+    Table updatedTable = existingTable.toBuilder()
```

Comment:
Clarification: We are creating the table by providing a schema without partition fields, but updating it with the schema by adding the partition fields, are we sure it's the right thing ? Without manifests and using ExternalTableDefinition, we use the schema without partition fields.

Reply:
Yes, when updating the schema, BigQuery expects all fields including the partition fields to be present.

Reply:
can you add java docs around this in lines 163 ish.